src/arrayMethodSort.js

```diff
@@ -5,7 +5,26 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    const comparator = compareFunction !== undefined
+      ? compareFunction
+      : (a, b) => String(a) > String(b) ? 1 : String(a) < String(b) ? -1 : 0;
```

Comment:
Don't use nested ternaries. It's a bad practice.
You can write sort function separately and use it as default in arguments.